src/initializers/bull/index.ts

```diff
@@ -0,0 +1,44 @@
+import { isEmpty } from 'lodash';
+import { parseURL } from 'ioredis/built/utils';
+import { getLogger } from '../log4js';
+import { OrkaOptions } from '../../typings/orka';
+
+let bull: any;
+
+export default async (config, orkaOptions: Partial<OrkaOptions>) => {
+  if (!config.bull || !config.bull.queue?.queues || (!config.bull.redis?.url && !config.redis?.url)) {
+    return;
+  }
+
+  const prefix = orkaOptions.appName;
+  const defaultOptions = config.bull.queue.options;
+  const redis = config.bull.redis || config.redis;
+  const queues = config.bull.queue.queues;
+
+  if (config.bull.redis?.tls) {
+    ['key', 'ca', 'cert'].forEach(prop => {
+      if (isEmpty(config.bull.redis.tls[prop])) delete config.bull.redis.tls[prop];
+    });
+    if (isEmpty(config.bull.redis.tls)) delete config.bull.redis.tls;
+  }
+  const redisOpts = {
+    ...parseURL(redis.url),
+    tls: redis.tls || redis.options?.tls,
+    enableReadyCheck: false
+  };
+  const Bull = (await import('./bull')).default;
+  bull = new Bull(prefix, queues, defaultOptions, redisOpts);
+  getLogger('orka.initializers.bull').info(
```

Comment:
Here as well, move the getLogger statements outside the function scope. Supposedly loggers are cached, but there is no need to invoke this constantly.

Reply:
but of course! map accepts a `thisArg`!
ty! 🙇